### PR TITLE
Set up CI/CD

### PR DESCRIPTION
**Pull request by hoangmirs**
https://github.com/hoangmirs/go-scraper/issues/3
https://github.com/hoangmirs/go-scraper/issues/4

## What happened 👀
- Set up CI/CD
- Deploy to Heroku by using Docker
- ~~NGINX support~~
 
## Insight 📝

- Test workflow is run on the normal go environment
- Deploy app with Docker
- ~~Modify to support NGINX~~
 
## Proof Of Work 📹

Can access this staging server: https://go-scraper-staging.herokuapp.com/

### CODE REVIEW

File: bin/start [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+
+sed -i -e 's/$PORT/'"$PORT"'/g' /etc/nginx/conf.d/default.conf
+
+nginx -c /etc/nginx/conf.d/default.conf
```
Comment (olivierobert) [inline bin/start:7]: Why is Nginx required for Heroku?

Also, getting 502 for https://go-scraper-staging.herokuapp.com/ 🤔 

image

Comment (hoangmirs) [reply bin/start]: I added it to serve the static files, but it's not necessary. Removed it in https://github.com/hoangmirs/go-scraper/pull/22/commits/2188b86275a4363182997b4257d45b92c95b1c94

**Review by olivierobert — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -8,21 +8,26 @@ COPY ./assets/. ./assets/
 
 RUN npm install && npm run build
 
-FROM golang:1.15-buster
 
-ENV GO111MOD=on
+FROM golang:1.15-buster as go-builder
```
Comment (Lahphim) [inline Dockerfile:12]: @hoangmirs Do you plan to use `-alpine` instead of `-buster` in this PR?

File: Dockerfile [hunk 3]
```diff
@@ -8,21 +8,26 @@ COPY ./assets/. ./assets/
...
-  go get github.com/beego/bee
+ENV GO111MOD=on
 
 WORKDIR /app
 
 COPY . .
 
 RUN go mod download
 
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build
```
Comment (Lahphim) [inline Dockerfile:22]: @hoangmirs Should we group these variables `CGO_ENABLED=0 GOOS=linux GOARCH=amd64` with this [line](https://github.com/hoangmirs/go-scraper/pull/22/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R14)?
So we can leave `RUN go build` standalone ✌️

**Review by Lahphim — APPROVED** [review]

**Pull request merged**